siddhi-testsuite/src/main/resources/archetype-resources/component/src/test/java/sample/AbstractTemperatureAlertTests.java

```diff
@@ -0,0 +1,152 @@
+package sample;
+
+import io.siddhi.core.SiddhiAppRuntime;
+import io.siddhi.core.SiddhiManager;
+import io.siddhi.core.stream.input.InputHandler;
+import io.siddhi.distribution.test.framework.MySQLContainer;
+import io.siddhi.distribution.test.framework.NatsContainer;
+import io.siddhi.distribution.test.framework.SiddhiRunnerContainer;
+import io.siddhi.distribution.test.framework.util.NatsClient;
+import io.siddhi.extension.io.nats.sink.NATSSink;
+import io.siddhi.extension.map.json.sinkmapper.JsonSinkMapper;
+import org.testcontainers.containers.output.WaitingConsumer;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Class for AbstractTemperatureAlertTests of Temp-Alert-App. This class includes the common logic for integration
+ * tests and blackbox tests.
+ *
+ * Description: Used for temperature monitoring and anomaly detection. Consumes events from a Nats topic,
+ *              filters the event under types 'monitored' and 'internal'.
+ *              Monitored events are then sent through a pattern and the matched events will be alerted to a Nats topic.
+ *              The internal events are persisted to a table.
+ */
+public abstract class AbstractTemperatureAlertTests {
+
+    public WaitingConsumer siddhiLogConsumer = new WaitingConsumer();
+
+    public MySQLContainer mySQLContainer;
```

Comment:
Why these containers are in Abstract class. These are only used in Integration tests?